purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -528,6 +566,66 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         debugLog("Invalidating Purchaser info cache")
         deviceCache.clearPurchaserInfoCacheTimestamp()
     }
+
+    // region Subscriber Attributes
+
+    /**
+     * Subscriber attributes are useful for storing additional, structured information on a user.
+     * Since attributes are writable using a public key they should not be used for
+     * managing secure or sensitive information such as subscription status, coins, etc.
+     *
+     * Key names starting with "$" are reserved names used by RevenueCat. For a full list of key
+     * restrictions refer to our guide: https://docs.revenuecat.com/docs/subscriber-attributes
+     *
+     * @param attributes Map of attributes by key. Set the value as null to delete an attribute.
+     */
+    fun setAttributes(attributes: Map<String, String?>) {
```

Comment:
[not for this PR]: 
thought: since Purchases.kt is kind of a big file, but devs might want to consult it for documentation on methods, would it make sense to extract an interface so that it only has the public methods (kinda like a header file), so devs only look at that one for reference?

Reply:
the advantage being that the interface would probably be like ~300 lines of code vs thousands, and it'd only contain the public stuff

Reply:
yes totally agree. I actually thought about doing a java interface and a kotlin interface to offer better cross language support